x-pack/plugins/alerting/server/task_runner/task_runner.ts

```diff
@@ -325,6 +333,16 @@ export class TaskRunner<
       searchSourceClient,
     });
 
+    const alertFactory = createAlertFactory<
```

Comment:
I found it difficult to get all the mocks right for unit testing the scenario where the circuit breaker is hit in the task runner. Changes to the `processAlerts` function and the alert factory are unit tested, so relying on functional tests to test what happens across multiple rule runs.